### ISSUE TITLE
Effect of fragment shader which writes to non-existent attachments.

### ISSUE THREAD

**Issue by NicolBolas** [KhronosGroup/Vulkan-Docs#198]
It is not clear from the Vulkan specification what the effect of a fragment shader is if a fragment shader writes to a `Location`, but there is no color attachment in the subpass for that location. All it says is:

> A fragment shader output variable identified with a Location decoration of `i` is directed to the color attachment indicated by `pColorAttachments[i]`

It says nothing about what happens if `pColorAttachments[i]` for that stage has no image attached. Presumably the results for that output will be discarded. But the main point is that it is not clear that this case represents valid usage.

**Comment by NicolBolas** [comment]
Any progress on this? Or did I miss some part of the specification where it explains the behavior?

**Comment by krOoze** [comment]
> When the fragment stage is present in a pipeline, the fragment shader outputs form an interface with the output attachments of the current subpass. The fragment shader output variables are matched by the Location and Component decorations to the color attachments specified in the pColorAttachments array of the VkSubpassDescription structure that describes the subpass that the fragment shader is executed in.

So, I think, that is a long winded explanation that such a action would break interface.

**Comment by jeffbolznv** [comment]
IIRC the intent is that the writes would be ignored. The renderpass chapter says:

```
    If any color or depth/stencil attachments are
    ename:VK_ATTACHMENT_UNUSED, then no writes occur for those attachments.
```

This should probably be clarified in the fragment output interface section.

**Comment by krOoze** [comment]
Well there are two cases in the OP: "there is no color attachment" and "pColorAttachments[i] for that stage has no image attached"...

**Comment by oddhack** [comment]
This should be fixed in the 1.0.16 spec update.
